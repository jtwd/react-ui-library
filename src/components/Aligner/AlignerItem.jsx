import React from 'react'
import { node, number } from 'prop-types'
import styled from 'styled-components'

const StyledContainer = styled.div`
  flex: ${props => (props.flex > 0) ? props.flex : 'none'};
`

/** Flexbox layout helper for alignment */
function AlignerItem({ children, flex, ...props}) {
  return (
    <StyledContainer flex={flex} {...props}>{children}</StyledContainer>
  )
}

AlignerItem.propTypes = {
  children: node.isRequired,
  flex: number
}

AlignerItem.defaultProps = {
  flex: 'none'
}

export default AlignerItem
