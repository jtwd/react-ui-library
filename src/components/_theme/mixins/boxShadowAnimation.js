import { borderShadow, borderShadowRaised, transitions } from "../units"

function boxShadowAnimation(action) {
  return `
    box-shadow: ${borderShadow};
    transition: ${transitions.hover};
    ${action === 'hover' && `
      &:hover {
        box-shadow: ${borderShadowRaised};
      }
    `}
    ${action === 'focus' && `
      &:focus {
        box-shadow: ${borderShadowRaised};
      }
    `}
    ${action === 'all' && `
      &:focus,
      &:hover {
        box-shadow: ${borderShadowRaised};
      }
    `}
  `
}

export default boxShadowAnimation
