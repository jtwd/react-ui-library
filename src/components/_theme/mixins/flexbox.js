export const packToFlex = {
  start: 'flex-start',
  center: 'center',
  end: 'flex-end',
  between: 'space-between',
  around: 'space-around'
}

export const alignToFlex = {
  stretch: 'stretch',
  top: 'flex-start',
  bottom: 'flex-end',
  center: 'center'
}
