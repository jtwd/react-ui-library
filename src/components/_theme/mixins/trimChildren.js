/**
 * Returns a style string to trim margin from children either horizontally or vertically
 * @param {string} dir - Direction to trim - 'hor'
 * @returns {string} - A style string for styled-components
 */
export default function trimChildren (dir = 'hor') {
  if (dir === 'ver') {
    return `
      &:first-child {
        margin-top: 0;
      }
      &:last-child {
        margin-bottom: 0;
      }
    `
  }
  if (dir === 'hor') {
    return `
      &:first-child {
        margin-left: 0;
      }
      &:last-child {
        margin-right: 0;
      }
    `
  }
  return `
    &:first-child {
      margin-top: 0;
      margin-left: 0;
    }
    &:last-child {
      margin-bottom: 0;
      margin-right: 0;
    }
  `
}