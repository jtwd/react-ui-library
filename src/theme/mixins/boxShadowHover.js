import { borderShadow, borderShadowRaised, transTextHover } from "../units"

const boxShadowHover = `
  box-shadow: ${borderShadow};
  transition: ${transTextHover};
  
  &:hover {
    box-shadow: ${borderShadowRaised};
  }
`

export default boxShadowHover