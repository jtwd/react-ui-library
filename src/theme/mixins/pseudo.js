export default function pseudo(display = 'block', pos = 'absolute', content = '') {
  return `
    content: ${content};
    display: ${display};
    position: ${pos};
  `
}