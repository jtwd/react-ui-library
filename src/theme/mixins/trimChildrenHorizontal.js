const trimChildrenHorizontal = `
  &:first-child {
    margin-left: 0;
  }
  &:last-child {
    margin-right: 0;
  }
`

export default trimChildrenHorizontal