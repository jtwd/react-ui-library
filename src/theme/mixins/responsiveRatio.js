import pseudo from './pseudo'

export default function responsiveRatio(x, y, usePseudo = false) {
  const padding = (y / x) + 100

  if (usePseudo) {
    return `
      &::before {
        ${pseudo('block', 'relative')}
        width: 100%;
        padding-top: ${padding}%;
      }
    `
  }
  return `
    padding-top: ${padding}%;
  `
}