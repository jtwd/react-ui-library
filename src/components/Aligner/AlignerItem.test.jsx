import React from 'react'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import AlignerItem from './AlignerItem'
import {alignToFlex} from "../_theme/mixins/flexbox"

describe('AlignerItem component', () => {
  it('renders correctly by default', () => {
    const component = shallow(<AlignerItem><div>Aligned item</div></AlignerItem>)
    expect(component).toMatchSnapshot()
  })

  it('renders correctly with a flex prop', () => {
    const component = shallow(<AlignerItem flex="1"><div>Aligned item</div></AlignerItem>)
    expect(component).toMatchSnapshot()
  })

  it('renders correctly with a flex prop of 0', () => {
    const Component = (
      <AlignerItem flex="0">
        <div>Aligned item</div>
      </AlignerItem>
    )
    const wrapper = renderer.create(Component).toJSON()
    expect(wrapper).toHaveStyleRule('flex', 'none')
  })

  it('renders correctly with a alignSelf prop of top', () => {
    const Component = (
      <AlignerItem alignSelf="top">
        <div>Aligned item</div>
      </AlignerItem>
    )
    const wrapper = renderer.create(Component).toJSON()
    expect(wrapper).toHaveStyleRule('align-self', alignToFlex.top)
  })

  it('renders correctly with a alignSelf prop of center', () => {
    const Component = (
      <AlignerItem alignSelf="center">
        <div>Aligned item</div>
      </AlignerItem>
    )
    const wrapper = renderer.create(Component).toJSON()
    expect(wrapper).toHaveStyleRule('align-self', alignToFlex.center)
  })

  it('renders correctly with a alignSelf prop of bottom', () => {
    const Component = (
      <AlignerItem alignSelf="bottom">
        <div>Aligned item</div>
      </AlignerItem>
    )
    const wrapper = renderer.create(Component).toJSON()
    expect(wrapper).toHaveStyleRule('align-self',  alignToFlex.bottom)
  })
})