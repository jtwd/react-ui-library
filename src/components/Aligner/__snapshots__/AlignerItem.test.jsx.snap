// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignerItem component renders correctly by default 1`] = `
<styled.div
  alignSelf={null}
  flex="none"
>
  <div>
    Aligned item
  </div>
</styled.div>
`;

exports[`AlignerItem component renders correctly with a flex prop 1`] = `
<styled.div
  alignSelf={null}
  flex="1"
>
  <div>
    Aligned item
  </div>
</styled.div>
`;
